osbs/api.py

```diff
@@ -405,12 +418,19 @@ def create_prod_build(self, git_uri, git_ref,
         :param architecture: str, build architecture
         :param yum_repourls: list, URLs for yum repos
         :param koji_task_id: int, koji task ID requesting build
+        :param filesystem_koji_task_id: int, koji task for base image filesystem
         :param scratch: bool, this is a scratch build
+        :param platform: str, the platform to build for
```

Comment:
Not necessarily (platform is more than the CPU architecture, it's also the host OS etc).

But more importantly 'architecture' is a label we definitely want to be set in the worker build, not in configuration, i.e. that label's value should ultimate come from querying the machine the build was run on.

We want 'platform' to come from the caller of this method, not from osbs.conf.